### PR TITLE
TEST/UCP/ERROR: Refactor test_ucp_peer_failure

### LINKED ISSUE #2537: Hang in dcx/test_ucp_peer_failure_2pairs.status_after_error/0
```
$ make -C test/gtest/ GTEST_FILTER="*test_ucp_peer_failure_2pairs.status_after_error*" test
```

RHEL7.5, OFED-internal-4.3-1.0.1, UCX master

dc/test_ucp_peer_failure_2pairs.status_after_error/0 test runs fine.

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2537 by increasing UD timeout

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7015/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7031/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7049/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7063/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7064/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7078/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7081/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7083/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7085/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7090/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7117/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_peer_failure.cc [hunk 1]
```diff
@@ -8,638 +8,400 @@
...
+    params.err_mode        = UCP_ERR_HANDLING_MODE_PEER;
+    params.err_handler.cb  = err_cb;
+    params.err_handler.arg = reinterpret_cast<void*>(this);
+    return params;
+}
 
-    fail_receiver();
+void test_ucp_peer_failure::set_timeouts() {
+    /* Set small TL timeouts to reduce testing time */
+    m_env.push_back(new ucs::scoped_setenv("UCX_RC_TIMEOUT",     "10ms"));
```
[the remark this answers is not in the document]
Comment (yosefe) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: RC_TIMEOUT is not in global opts, it's iface opts..

File: test/gtest/ucp/test_ucp_peer_failure.cc [hunk 2]
```diff
@@ -8,638 +8,400 @@
...
-        EXPECT_LT(allocd_eps_after, allocd_eps_before);
-    }
+ucp_test::entity& test_ucp_peer_failure::stable_receiver() {
+    return m_entities.at(m_entities.size() - 2);
+}
 
-    /* The EP can't be used now */
-    sender().revoke_ep();
+ucp_test::entity& test_ucp_peer_failure::failing_receiver() {
+    return m_entities.at(m_entities.size() - 1);
```
[the remark this answers is not in the document]
Comment (yosefe) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: wanted to make it consistent with stable_receiver()

[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: Ok

File: test/gtest/ucp/test_ucp_peer_failure.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: Ok

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7120/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7121/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov fixed the comment

**Pull request merged**